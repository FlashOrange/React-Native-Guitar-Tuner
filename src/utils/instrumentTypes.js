const guitar = {
  displayName: 'Guitar',
  stringNotes: ['E', 'A', 'D', 'G', 'B', 'E'],
  stringFrequencies: [
    { min: 36, max: 96, tuned: 82 },
    { min: 96, max: 128, tuned: 110 },
    { min: 128, max: 171, tuned: 146 },
    { min: 171, max: 221, tuned: 196 },
    { min: 221, max: 287, tuned: 246 },
    { min: 287, max: 371, tuned: 329 },
  ],
};

const bass4 = {
  displayName: 'Bass (4-String)',
  stringNotes: ['E', 'A', 'D', 'G'],
  stringFrequencies: [
    { min: 30, max: 48.55, tuned: 41.2 },
    { min: 48.56, max: 64.21, tuned: 55 },
    { min: 64.22, max: 86.61, tuned: 73.42 },
    { min: 86.62, max: 110, tuned: 98 },
  ],
};

const bass5low = {
  displayName: 'Bass (5-String, Low-B)',
  stringNotes: ['B', 'E', 'A', 'D', 'G'],
  stringFrequencies: [
    { min: 20, max: 36.48, tuned: 30.87 },
    { min: 36.49, max: 48.55, tuned: 42.1 },
    { min: 48.56, max: 64.21, tuned: 55 },
    { min: 64.22, max: 86.61, tuned: 73.42 },
    { min: 86.62, max: 140, tuned: 98 },
  ],
};

const bass5hi = {
  displayName: 'Bass (5-String, High-C)',
  stringNotes: ['E', 'A', 'D', 'G', 'C'],
  stringFrequencies: [
    { min: 36.49, max: 48.55, tuned: 41.2 },
    { min: 48.56, max: 64.21, tuned: 55 },
    { min: 64.22, max: 86.61, tuned: 73.42 },
    { min: 86.62, max: 114.4, tuned: 98 },
    { min: 114.41, max: 150, tuned: 130.81 },
  ],
};

const ukulele = {
  displayName: 'Ukulele',
  stringNotes: ['G', 'C', 'E', 'A'],
  stringFrequencies: [
    { min: 360.85, max: 416, tuned: 392 },
    { min: 326.81, max: 295.36, tuned: 261.63 },
    { min: 295.37, max: 360.84, tuned: 329.63 },
    { min: 416.1, max: 114.4, tuned: 440 },
  ],
};

const instrumentTypes = {
  guitar,
  bass4,
  bass5low,
  bass5hi,
  ukulele,
};

export default instrumentTypes;
