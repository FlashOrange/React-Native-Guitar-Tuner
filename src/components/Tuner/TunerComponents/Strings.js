import React, { Component } from 'react';
import { StyleSheet, Image, View, Text } from 'react-native';
import PropTypes from 'prop-types';
import * as Animatable from 'react-native-animatable';

import { colors } from '../../../utils';

export default class Strings extends Component {
  render() {
    console.log(this.props.activeString);
    return (
      <View style={styles.stringsContainer}>
        {this.props.stringNotes.map((note, i) => {
          let image = null;
          if (this.props.activeString === i) {
            image = (
              <Animatable.Image
                animation="shake"
                duration={20}
                easing="ease-in-out-quint"
                iterationCount="infinite"
                source={require('../../../../img/string.jpg')}
                style={styles.string}
              />
            );
          } else {
            image = (
              <Image
                source={require('../../../../img/string.jpg')}
                style={styles.string}
              />
            );
          }
          return (
            <View key={i} style={styles.stringContainer}>
              {image}
              <View
                style={[
                  styles.noteContainer,
                  {
                    borderColor:
                      this.props.activeString === i
                        ? colors.colorPrimary
                        : colors.colorSecondary,
                    borderWidth: this.props.activeString === i ? 4 : 3,
                  },
                ]}
              >
                <Text style={styles.note}>{note}</Text>
              </View>
            </View>
          );
        })}
      </View>
    );
  }
}

Strings.propTypes = {
  activeString: PropTypes.number,
};

const styles = StyleSheet.create({
  stringsContainer: {
    borderTopColor: colors.colorSecondary,
    borderTopWidth: 5,
    justifyContent: 'space-around',
    flexDirection: 'row',
    flex: 1,
  },
  stringContainer: {
    alignItems: 'center',
  },
  string: {
    flex: 1,
  },
  note: {
    color: 'white',
    fontSize: 19,
    textAlign: 'center',
  },
  noteContainer: {
    top: 50,
    height: 50,
    width: 50,
    position: 'absolute',
    padding: 10,
    borderColor: colors.colorPrimary,
    borderWidth: 3,
    borderRadius: 25,
    backgroundColor: colors.background,
  },
});
