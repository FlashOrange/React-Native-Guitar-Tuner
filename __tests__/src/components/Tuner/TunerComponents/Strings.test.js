import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import instrumentTypes from '../../../../../src/utils/instrumentTypes';
import { shallow } from 'enzyme';
import Strings from '../../../../../src/components/Tuner/TunerComponents/Strings';
import { Image as AnimatedImage } from 'react-native-animatable';
const instrument = instrumentTypes[Object.keys(instrumentTypes)[0]];

const config = {
  stringNotes: instrument.stringNotes,
  activeString: null,
};

describe('<Strings />', () => {
  const Component = <Strings {...config} />;
  let strings;

  it('renders correctly', () => {
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  beforeEach(() => {
    strings = shallow(Component);
  });

  it('renders the correct number of strings', () => {
    const images = strings.find('Image');
    expect(images.length).toEqual(instrument.stringNotes.length);
  });

  it('displays each note name', () => {
    const text = strings.find('Text');
    const notes = text.map(t => {
      return t.children().text();
    });
    expect(notes).toEqual(instrument.stringNotes);
  });

  it('animates the active string', () => {
    const clone = React.cloneElement(Component, { activeString: 0 });
    const stringsActivated = shallow(clone);
    const animatable = stringsActivated.find(AnimatedImage);
    expect(animatable.length).toEqual(1);
    expect(stringsActivated.childAt(0).childAt(0)).toEqual(animatable);
  });
});
