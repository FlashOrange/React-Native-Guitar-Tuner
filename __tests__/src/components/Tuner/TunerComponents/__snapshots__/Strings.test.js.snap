// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Strings /> renders correctly 1`] = `
<View
  style={
    Object {
      "borderTopColor": "#81D4FA",
      "borderTopWidth": 5,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../img/string.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1f2025",
            "borderColor": "#FFD54F",
            "borderRadius": 25,
            "borderWidth": 3,
            "height": 50,
            "padding": 10,
            "position": "absolute",
            "top": 50,
            "width": 50,
          },
          Object {
            "borderColor": "#81D4FA",
            "borderWidth": 3,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 19,
            "textAlign": "center",
          }
        }
      >
        E
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../img/string.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1f2025",
            "borderColor": "#FFD54F",
            "borderRadius": 25,
            "borderWidth": 3,
            "height": 50,
            "padding": 10,
            "position": "absolute",
            "top": 50,
            "width": 50,
          },
          Object {
            "borderColor": "#81D4FA",
            "borderWidth": 3,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 19,
            "textAlign": "center",
          }
        }
      >
        A
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../img/string.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1f2025",
            "borderColor": "#FFD54F",
            "borderRadius": 25,
            "borderWidth": 3,
            "height": 50,
            "padding": 10,
            "position": "absolute",
            "top": 50,
            "width": 50,
          },
          Object {
            "borderColor": "#81D4FA",
            "borderWidth": 3,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 19,
            "textAlign": "center",
          }
        }
      >
        D
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../img/string.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1f2025",
            "borderColor": "#FFD54F",
            "borderRadius": 25,
            "borderWidth": 3,
            "height": 50,
            "padding": 10,
            "position": "absolute",
            "top": 50,
            "width": 50,
          },
          Object {
            "borderColor": "#81D4FA",
            "borderWidth": 3,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 19,
            "textAlign": "center",
          }
        }
      >
        G
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../img/string.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1f2025",
            "borderColor": "#FFD54F",
            "borderRadius": 25,
            "borderWidth": 3,
            "height": 50,
            "padding": 10,
            "position": "absolute",
            "top": 50,
            "width": 50,
          },
          Object {
            "borderColor": "#81D4FA",
            "borderWidth": 3,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 19,
            "textAlign": "center",
          }
        }
      >
        B
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../img/string.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1f2025",
            "borderColor": "#FFD54F",
            "borderRadius": 25,
            "borderWidth": 3,
            "height": 50,
            "padding": 10,
            "position": "absolute",
            "top": 50,
            "width": 50,
          },
          Object {
            "borderColor": "#81D4FA",
            "borderWidth": 3,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 19,
            "textAlign": "center",
          }
        }
      >
        E
      </Text>
    </View>
  </View>
</View>
`;
